insets/src/main/java/com/google/accompanist/insets/Insets.kt

```diff
@@ -45,93 +47,196 @@ class WindowInsets {
     /**
      * Inset values which match [WindowInsetsCompat.Type.systemBars]
      */
-    val systemBars = Insets()
+    val systemBars: InsetsType = InsetsType()
 
     /**
      * Inset values which match [WindowInsetsCompat.Type.systemGestures]
      */
-    val systemGestures = Insets()
+    val systemGestures: InsetsType = InsetsType()
 
     /**
      * Inset values which match [WindowInsetsCompat.Type.navigationBars]
      */
-    val navigationBars = Insets()
+    val navigationBars: InsetsType = InsetsType()
 
     /**
      * Inset values which match [WindowInsetsCompat.Type.statusBars]
      */
-    val statusBars = Insets()
+    val statusBars: InsetsType = InsetsType()
 
     /**
      * Inset values which match [WindowInsetsCompat.Type.ime]
      */
-    val ime = Insets()
+    val ime: InsetsType = InsetsType()
 }
 
+/**
+ * Represents the values for a type of insets, and stored information about the layout insets,
+ * animating insets, and visibility of the insets.
+ *
+ * [InsetsType] instances are commonly stored in a [WindowInsets] instance.
+ */
 @Stable
-class Insets {
+@Suppress("MemberVisibilityCanBePrivate")
+class InsetsType : Insets {
+    private var ongoingAnimations by mutableStateOf(0)
+    internal val _layoutInsets = MutableInsets()
+    internal val _animatedInsets = MutableInsets()
+
     /**
-     * The left dimension of these insets in pixels.
+     * The layout insets for this [InsetsType]. These are the insets which are defined from the
+     * current window layout.
+     *
+     * You should not normally need to use this directly, and instead use [left], [top],
+     * [right], and [bottom] to return the correct value for the current state.
      */
-    var left by mutableStateOf(0)
-        internal set
+    val layoutInsets: Insets
```

Comment:
Do you need to expose both `layoutInsets` and `animatedInsets`? What about keeping those private to this class and only exposing a single `currentInsets` and public methods to update layout/animated insets?

Reply:
Yeah, that was the idea behind the request (see the Slack chat linked in #242). Basically they want to manually read only the layout insets + animated fraction. Seems like a good FR to me, and is the exact same information which the platform provides.

Reply:
Ahh ack. Sorry didn't read the issue.